### ISSUE TITLE
False positive PLC0208

### ISSUE THREAD

**Issue by frenck** [astral-sh/ruff#5386]
There is a bug / false positive in rule PLC0208: `iteration-over-set`.

According to the description: "Checks for iterations over set literals".

However, it does trigger when not literals are not used.

Examples code from our codebase:

```python
for entity_id in {
    *self.hass.states.async_entity_ids(),
    *self._prefs.google_entity_configs,
}:
   ...
```

Results in: ```Use a sequence type instead of a `set` when iterating over values```

Current ruff version 0.0.272

Came up during the review of extending Ruff rulesets, the follow PR might provide more context to this issue: <https://github.com/home-assistant/core/pull/94359#pullrequestreview-1473654844>

Verified that pylint doesn't trigger on these cases.

../Frenck

**Comment by charliermarsh** [comment]
Thanks!

**Comment by frenck** [comment]
Awesome! Thanks for the quick turn-around ❤️
